akka-http-core/src/main/scala/akka/http/impl/util/package.scala

```diff
@@ -113,6 +105,27 @@ package util {
   import akka.stream.{ Attributes, Outlet, Inlet, FlowShape }
   import scala.concurrent.duration.FiniteDuration
 
+  private[http] final case class ErrorHandling[T](handler: Throwable ⇒ Unit) extends GraphStage[FlowShape[T, T]] {
+    private val in = Inlet[T]("ErrorHandling.in")
+    private val out = Outlet[T]("ErrorHandling.out")
+
+    override val shape = FlowShape(in, out)
+
+    override def createLogic(attr: Attributes) =
+      new GraphStageLogic(shape) with InHandler with OutHandler {
```

Comment:
Good, we want the http pipeline to be as lean on allocations as possible, so :+1: on that trick.